### PR TITLE
Update Kotlin Version

### PR DESCRIPTION
**Pull request by kidinov**
### Description

The latest version of Stripe requires kotlin 1.8.10. This PR brings this update so we can slightly reduce risk by updating it transitively

### Testing instructions

A general smoke test is required. It updates compose, so more attention to compose screens is nessasery

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/moremenu/MoreMenuScreen.kt [hunk 1]
```diff
@@ -380,6 +380,8 @@ fun MoreMenuBadge(badgeState: BadgeState?) {
             verticalAlignment = Alignment.CenterVertically
         ) {
             Spacer(modifier = Modifier.weight(1f))
+
+            @Suppress("RememberReturnType")
```
Comment (kidinov) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/moremenu/MoreMenuScreen.kt:384]: False positive caused by [the bug](https://issuetracker.google.com/issues/263887242)

**Comment by wzieba** [comment]
You might want to take a look at this PR: https://github.com/woocommerce/woocommerce-android/pull/8691. There's a comment why we might need to update the compose compiler to `1.4.6`. 

Edit: or maybe we don't have to, as it's an update to 1.8.10 and not 1.8.20 🤔 .

**Comment by kidinov** [comment]
@wzieba I was following the [compatibility map](https://developer.android.com/jetpack/androidx/releases/compose-kotlin)

Should I close this PR as it pretty much duplicates the one you linked?

**Comment by wzieba** [comment]
> Should I close this PR as it pretty much duplicates the one you linked?

I think it's okay to proceed with this PR as it updates Kotlin to 1.8.10, as you already resolved some issues that are present on the linked PR.

We can work on bumping to 1.8.20 later to not block you with Stripe SDK update.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/moremenu/MoreMenuScreen.kt (hunk 1 above)
Comment (wzieba) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/moremenu/MoreMenuScreen.kt:384]: Do you know which version will include the fix? I've been trying to find it on attached link but couldn't.

**Review by wzieba — APPROVED** [review]
Looks good to me - I've smoked tested the app, especially the compose screens, and haven't found any issues.

**Pull request merged**